huaweicloud/services/swr/resource_huaweicloud_swr_domain.go

```diff
@@ -0,0 +1,192 @@
+	}
+
+	organization := d.Get("organization").(string)
+	repository := d.Get("organization").(string)
+
+	err = domains.Create(client, organization, repository, opts).ExtractErr()
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud SWR domain: %w", err)
+	}
+	d.SetId(domain)
+
+	return resourceSWRDomainRead(ctx, d, meta)
+}
+
+func resourceSWRDomainRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.SwrV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to create HuaweiCloud SWR client : %s", err)
+	}
+
+	organization := d.Get("organization").(string)
+	repository := d.Get("organization").(string)
+
+	domain, err := domains.Get(client, organization, repository, d.Id()).Extract()
+	if err != nil {
+		return fmtp.DiagErrorf("Error retrieving HuaweiCloud SWR domain: %w", err)
+	}
+
+	mErr := multierror.Append(
+		d.Set("region", config.GetRegion(d)),
+		d.Set("access_domain", domain.AccessDomain),
+		d.Set("repository", domain.Repository),
+		d.Set("organization", domain.Organization),
+		d.Set("description", domain.Description),
+		d.Set("status", domain.Status),
+		d.Set("permission", domain.Permit),
+		d.Set("created", domain.Created),
+		d.Set("updated", domain.Updated),
+		d.Set("creator_id", domain.CreatorID),
+		d.Set("creator_name", domain.CreatorName),
+	)
+	if err := mErr.ErrorOrNil(); err != nil {
+		return fmtp.DiagErrorf("error setting HuaweiCloud SWR Domain fields: %w", err)
+	}
+
+	return nil
+}
+
+func resourceSWRDomainUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.SwrV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to create HuaweiCloud SWR client : %s", err)
+	}
+
+	description := d.Get("description").(string)
+	opts := domains.UpdateOpts{
+		Permit:      d.Get("permission").(string),
+		Deadline:    d.Get("deadline").(string),
+		Description: &description,
+	}
+
+	organization := d.Get("organization").(string)
+	repository := d.Get("organization").(string)
+
+	err = domains.Update(client, organization, repository, d.Id(), opts).ExtractErr()
+	if err != nil {
+		return fmtp.DiagErrorf("Error updating HuaweiCloud SWR Domain: %w", err)
+	}
+
+	return resourceSWRDomainRead(ctx, d, meta)
+}
+
+func resourceSWRDomainDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.SwrV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to create HuaweiCloud SWR client : %s", err)
+	}
+
+	organization := d.Get("organization").(string)
+	repository := d.Get("organization").(string)
+
+	err = domains.Delete(client, organization, repository, d.Id()).ExtractErr()
+	if err != nil {
+		fmtp.DiagErrorf("error deleting HuaweiCloud SWR domain: %w", err)
```

Comment:
maybe use `common.CheckDeletedDiag` is better

Reply:
fixed